gammapy/estimators/ts_map.py

```diff
@@ -637,18 +663,30 @@ def ts_diff(x):
             max_norm = norm
             factor = -1
 
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
+        try:
             roots, res = find_roots(
-                ts_diff,
-                [min_norm],
-                [max_norm],
-                nbin=1,
-                maxiter=self.max_niter,
-                rtol=self.rtol,
-            )
-            # Where the root finding fails NaN is set as norm
-            return (roots[0] - norm) * factor
+                    ts_diff,
+                    [min_norm],
+                    [max_norm],
+                    nbin=1,
+                    maxiter=self.max_niter,
+                    rtol=self.rtol,
+                )
+            value = (roots[0] - norm) * factor
+            if "converged" not in res[0].flag:
+                value = np.nan
```

Comment:
according to the previous comment "Where the root finding fails NaN is set as norm" then I guess that roots[0] is already NaN so the extra if is not necessary.